lib/protobuf/json.ex

```diff
@@ -0,0 +1,161 @@
+  as [well-known](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf)
+  types are not fully supported yet.
+
+  ## Types
+
+  Protobuf messages and embedded messages are encoded as objects, _repeated_ fields as arrays,
+  _bytes_ as Base64 strings, _bool_ as booleans, _map_ as objects and so on.
+
+  | proto3                   | JSON                  | Supported |
+  |--------------------------|-----------------------|-----------|
+  | `message`                | `object`              | Yes       |
+  | `enum`                   | `string`              | Yes       |
+  | `map<K,V>`               | `object`              | Yes       |
+  | `repeated V`             | `array [v, …]`        | Yes       |
+  | `bool`                   | `true, false`         | Yes       |
+  | `string`                 | `string`              | Yes       |
+  | `bytes`                  | `base64 string`       | Yes       |
+  | `int32, fixed32, uint32` | `number`              | Yes       |
+  | `int64, fixed64, uint64` | `string`              | Yes       |
+  | `float, double`          | `number`              | Yes       |
+  | `Any`                    | `object`              | No        |
+  | `Timestamp`              | `string`              | No        |
+  | `Duration`               | `string`              | No        |
+  | `Struct`                 | `object`              | No        |
+  | `Wrapper types`          | `various types`       | No        |
+  | `FieldMask`              | `string`              | No        |
+  | `ListValue`              | `array [foo, bar, …]` | No        |
+  | `Value`                  | `value`               | No        |
+  | `NullValue`              | `null`                | No        |
+  | `Empty`                  | `object`              | No        |
+
+  ## Usage
+
+  `Protobuf.JSON` requires a JSON library to work, so first make sure you have `:jason` added
+  to your dependencies:
+
+      defp deps do
+        [
+          {:jason, "~> 1.2"},
+          # ...
+        ]
+      end
+
+  With `encode/1` you can turn any `Protobuf` message struct into a JSON string:
+
+      iex> message = Car.new(color: :RED, top_speed: 125.3)
+      iex> Protobuf.JSON.encode(message)
+      {:ok, "{\\"color\\":\\"RED\\",\\"topSpeed\\":125.3}"}
+
+  JSON keys are encoded as camelCase strings, specified by the `json_name` field options. So
+  make sure to recompile the `.proto` files in your project before working with JSON encoding.
+  You can set your own `json_name` too:
+
+      message GeoCoordinate {
+        double latitude = 1 [ json_name = "lat" ];
+        double longitude = 2 [ json_name = "long" ];
+      }
+
+  """
+
+  alias Protobuf.JSON.{Encode, EncodeError}
+
+  @type encode_opt :: {:json_library, module} | Encode.encode_opt()
+
+  @doc """
+  Generates a JSON representation of the given protobuf `struct`.
+
+  Similar to `encode/1` except it will unwrap the error tuple and raise in case of errors.
```

Comment:
```suggestion
  Similar to `encode/2` except it will unwrap the error tuple and raise in case of errors.
  Accepts the same options as `encode/2`.
```